datafusion/core/tests/user_defined/relation_planner.rs

```diff
@@ -0,0 +1,353 @@
+        _context: &mut dyn RelationPlannerContext,
+    ) -> Result<RelationPlanning> {
+        match relation {
+            TableFactor::Table { name, alias, .. }
+                if name.to_string().eq_ignore_ascii_case("colors") =>
+            {
+                let plan = LogicalPlanBuilder::values(vec![
+                    vec![Expr::Literal(ScalarValue::Utf8(Some("red".into())), None)],
+                    vec![Expr::Literal(ScalarValue::Utf8(Some("green".into())), None)],
+                    vec![Expr::Literal(ScalarValue::Utf8(Some("blue".into())), None)],
+                ])?
+                .build()?;
+                Ok(RelationPlanning::Planned(PlannedRelation::new(plan, alias)))
+            }
+            other => Ok(RelationPlanning::Original(other)),
+        }
+    }
+}
+
+/// A planner that intercepts nested joins and plans them recursively
+#[derive(Debug)]
+struct RecursiveJoinPlanner;
+
+impl RelationPlanner for RecursiveJoinPlanner {
+    fn plan_relation(
+        &self,
+        relation: TableFactor,
+        context: &mut dyn RelationPlannerContext,
+    ) -> Result<RelationPlanning> {
+        match relation {
+            TableFactor::NestedJoin {
+                table_with_joins,
+                alias,
+                ..
+            } if table_with_joins.joins.len() == 1 => {
+                // Recursively plan both sides using context.plan()
+                let left = context.plan(table_with_joins.relation.clone())?;
+                let right = context.plan(table_with_joins.joins[0].relation.clone())?;
+
+                // Create a cross join
+                let plan = LogicalPlanBuilder::from(left).cross_join(right)?.build()?;
+                Ok(RelationPlanning::Planned(PlannedRelation::new(plan, alias)))
+            }
+            other => Ok(RelationPlanning::Original(other)),
+        }
+    }
+}
+
+/// A planner that always returns None to test delegation
+#[derive(Debug)]
+struct PassThroughPlanner;
+
+impl RelationPlanner for PassThroughPlanner {
+    fn plan_relation(
+        &self,
+        relation: TableFactor,
+        _context: &mut dyn RelationPlannerContext,
+    ) -> Result<RelationPlanning> {
+        // Always return Original - delegates to next planner or default
+        Ok(RelationPlanning::Original(relation))
+    }
+}
+
+async fn collect_sql(ctx: &SessionContext, sql: &str) -> Vec<RecordBatch> {
+    ctx.sql(sql).await.unwrap().collect().await.unwrap()
+}
+
+#[tokio::test]
+async fn test_custom_planner_handles_relation() {
+    let ctx = SessionContext::new();
+    ctx.register_relation_planner(Arc::new(CustomValuesPlanner))
+        .unwrap();
+
+    let results = collect_sql(&ctx, "SELECT * FROM custom_values").await;
+
+    let expected = "\
++---------+
+| column1 |
++---------+
+| 1       |
+| 2       |
+| 3       |
++---------+";
+    assert_eq!(batches_to_string(&results), expected);
+}
+
+#[tokio::test]
+async fn test_multiple_planners_first_wins() {
```

Comment:
I do not get the name of this test. I do not see anything about `first wins`. Either you rename the test (which I think the case) or change the test for us to see there is a win